feat(webapp): 优化摄像头列表页面功能和样式

sunsulei opened the pull request:
- 添加删除摄像头功能
- 优化摄像头列表展示样式
- 增加用户确认和错误提示
- 调整摄像头加载逻辑
- 统一 API 请求处理
- 处理 403错误码，增加权限控制

Commit:
feat(webapp): 优化摄像头列表页面功能和样式

- 添加删除摄像头功能
- 优化摄像头列表展示样式
- 增加用户确认和错误提示
- 调整摄像头加载逻辑
- 统一 API 请求处理
- 处理 403错误码，增加权限控制

rkonfj commented on cmd/nvrd/response.go:
```diff
@@ -33,5 +33,5 @@ func Ok(data any) Data {
 
 func ErrForbidden(err error) errdefs.Error {
-	return errdefs.Error{Code: 4030, Msg: fmt.Errorf("forbidden: %w", err).Error()}
+	return errdefs.Error{Code: 403, Msg: fmt.Errorf("forbidden: %w", err).Error()}
```
Code 改回 4030

sunsulei replied:
已改回

rkonfj commented on webapp/src/http.js:
```diff
@@ -57,6 +55,11 @@ async function checkResp(resp) {
   }
+
+  if (resp.status === 403) {
+    window.location.href = '/'
```
这里不能使用 window 相关 api。抛出异常调用方处理

sunsulei replied:
去掉了这个if

rkonfj commented:
非常棒！

处理下提到的问题，再用 prettier 格式化一下。`npm run format`

rkonfj commented on webapp/src/views/Cameras.vue:
```diff
@@ -19,28 +19,30 @@ const passwordRef = ref()
+watch(cameras, n => {
+  console.log('cameras: ', n)
+}, {immediate: true})
+
```
这块是干啥的，是否可以去掉？

sunsulei replied:
去掉了.

Commit:
refactor(webapp): 优化代码结构和 API 服务器配置

- 移除watch
- 改回403
- prettier format

sunsulei committed:
Merge branch 'sigcn:main' into main

Commit:
feat: 初始化 NVR React 项目

- 创建基本的项目结构和配置文件
- 添加全局样式和 TailwindCSS 配置
- 实现简单的导航栏和布局组件
- 创建几个基本的页面组件
- 添加请求拦截器和全局消息处理工具
- 配置 Dockerfile 用于构建和部署

Commit:
chore: 更新 .gitignore 文件

- 添加 webapp_react目录下的 node_modules、build 和 dist 文件夹到忽略列表
- 保持 .idea 文件夹继续被忽略
